plenum/test/test_stack.py

```diff
@@ -29,6 +31,26 @@ def resetDelays(self):
         self.stasher.resetDelays()
 
 
+class TestKITStack(KITStack):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.declinedJoins = 0
+
+    def processRx(self, packet):
+        # Override to add check that in case of join new remote is in registry. This is done to avoid creation
+        # of unnecessary JSON files for remotes
+        tk = packet.data['tk']
+
+        if tk in [TrnsKind.join]:  # join transaction
+            sha = (packet.data['sh'], packet.data['sp'])
+            if not super().findInNodeRegByHA(sha):
+                logger.debug('Remote with HA {} not found in registry'.format(sha))
+                self.declinedJoins += 1
+                return
+
+        return super().processRx(packet)
+
```

Comment:
This practice has become an antipattern in my view. Subclassing classes with Test versions of those classes is messy, and promotes behaviors that can mask bugs. Suggest using Spyable. Put the logic of line 47 and 48 in another method and register it with Spyable such that you can count the number of times that method has been called. Much cleaner, and doesn't have the risks of subclassing.